// @chatbotkit/sdk/dist/cjs/integration/discord/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscordIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class DiscordIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listDiscordIntegrations)(this, request);
    }
    fetch(discordId) {
        return (0, v1_js_1.fetchDiscordIntegration)(this, discordId);
    }
    create(request) {
        return (0, v1_js_1.createDiscordIntegration)(this, request);
    }
    update(discordId, request) {
        return (0, v1_js_1.updateDiscordIntegration)(this, discordId, request);
    }
    delete(discordId) {
        return (0, v1_js_1.deleteDiscordIntegration)(this, discordId);
    }
    setup(discordId) {
        return (0, v1_js_1.setupDiscordIntegration)(this, discordId);
    }
}
exports.DiscordIntegrationClient = DiscordIntegrationClient;
exports.default = DiscordIntegrationClient;
