// @chatbotkit/sdk/dist/cjs/integration/telegram/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TelegramIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class TelegramIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listTelegramIntegrations)(this, request);
    }
    fetch(telegramId) {
        return (0, v1_js_1.fetchTelegramIntegration)(this, telegramId);
    }
    create(request) {
        return (0, v1_js_1.createTelegramIntegration)(this, request);
    }
    update(telegramId, request) {
        return (0, v1_js_1.updateTelegramIntegration)(this, telegramId, request);
    }
    delete(telegramId) {
        return (0, v1_js_1.deleteTelegramIntegration)(this, telegramId);
    }
    setup(telegramId) {
        return (0, v1_js_1.setupTelegramIntegration)(this, telegramId);
    }
}
exports.TelegramIntegrationClient = TelegramIntegrationClient;
exports.default = TelegramIntegrationClient;
