// @chatbotkit/sdk/dist/cjs/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatBotKit = exports.SecretClient = exports.ContactClient = exports.ConversationClient = exports.IntegrationClient = exports.SkillsetClient = exports.DatasetClient = exports.PartnerClient = exports.MagicClient = exports.FileClient = exports.BotClient = void 0;
const index_js_1 = require("./bot/index.cjs");
Object.defineProperty(exports, "BotClient", { enumerable: true, get: function () { return index_js_1.BotClient; } });
const client_js_1 = require("./client.cjs");
const index_js_2 = require("./contact/index.cjs");
Object.defineProperty(exports, "ContactClient", { enumerable: true, get: function () { return index_js_2.ContactClient; } });
const index_js_3 = require("./conversation/index.cjs");
Object.defineProperty(exports, "ConversationClient", { enumerable: true, get: function () { return index_js_3.ConversationClient; } });
const index_js_4 = require("./dataset/index.cjs");
Object.defineProperty(exports, "DatasetClient", { enumerable: true, get: function () { return index_js_4.DatasetClient; } });
const index_js_5 = require("./file/index.cjs");
Object.defineProperty(exports, "FileClient", { enumerable: true, get: function () { return index_js_5.FileClient; } });
const index_js_6 = require("./integration/index.cjs");
Object.defineProperty(exports, "IntegrationClient", { enumerable: true, get: function () { return index_js_6.IntegrationClient; } });
const index_js_7 = require("./magic/index.cjs");
Object.defineProperty(exports, "MagicClient", { enumerable: true, get: function () { return index_js_7.MagicClient; } });
const index_js_8 = require("./partner/index.cjs");
Object.defineProperty(exports, "PartnerClient", { enumerable: true, get: function () { return index_js_8.PartnerClient; } });
const index_js_9 = require("./secret/index.cjs");
Object.defineProperty(exports, "SecretClient", { enumerable: true, get: function () { return index_js_9.SecretClient; } });
const index_js_10 = require("./skillset/index.cjs");
Object.defineProperty(exports, "SkillsetClient", { enumerable: true, get: function () { return index_js_10.SkillsetClient; } });
class ChatBotKit extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
        this.bot = new index_js_1.BotClient(options);
        this.file = new index_js_5.FileClient(options);
        this.magic = new index_js_7.MagicClient(options);
        this.partner = new index_js_8.PartnerClient(options);
        this.dataset = new index_js_4.DatasetClient(options);
        this.skillset = new index_js_10.SkillsetClient(options);
        this.integration = new index_js_6.IntegrationClient(options);
        this.conversation = new index_js_3.ConversationClient(options);
        this.contact = new index_js_2.ContactClient(options);
        this.secrets = new index_js_9.SecretClient(options);
    }
}
exports.ChatBotKit = ChatBotKit;
exports.default = ChatBotKit;
