// @chatbotkit/sdk/dist/cjs/file/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.downloadFile = exports.uploadFile = exports.deleteFile = exports.updateFile = exports.createFile = exports.fetchFile = exports.listFiles = void 0;
const buffer_js_1 = require("../buffer.cjs");
function listFiles(client, request) {
    let url = `/api/v1/file/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listFiles = listFiles;
async function fetchFile(client, fileId) {
    const url = `/api/v1/file/${fileId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchFile = fetchFile;
async function createFile(client, request) {
    const url = `/api/v1/file/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createFile = createFile;
async function updateFile(client, fileId, request) {
    const url = `/api/v1/file/${fileId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateFile = updateFile;
async function deleteFile(client, fileId) {
    const url = `/api/v1/file/${fileId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteFile = deleteFile;
async function uploadFile(client, fileId, request) {
    const url = `/api/v1/file/${fileId}/upload`;
    const buffer = (0, buffer_js_1.getBuffer)(request.data);
    const response = await client.clientFetch(url, {
        record: {
            file: {
                size: buffer.byteLength,
                type: request.type,
                name: request.name,
            },
        },
    });
    const { uploadRequest } = response;
    if (!uploadRequest) {
        throw new Error('Missing upload request');
    }
    await client.clientFetch(uploadRequest.url, {
        method: uploadRequest.method,
        headers: uploadRequest.headers,
        buffer: buffer,
        external: true,
    });
    return {
        id: response.id,
    };
}
exports.uploadFile = uploadFile;
async function downloadFile(client, fileId) {
    const url = `/api/v1/file/${fileId}/download`;
    const response = client.clientFetch(url);
    return response;
}
exports.downloadFile = downloadFile;
