// @chatbotkit/sdk/dist/esm/integration/support/v1.js
export function listSupportIntegrations(client, request) {
    let url = `/api/v1/integration/support/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchSupportIntegration(client, supportId) {
    const url = `/api/v1/integration/support/${supportId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createSupportIntegration(client, request) {
    const url = `/api/v1/integration/support/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateSupportIntegration(client, supportId, request) {
    const url = `/api/v1/integration/support/${supportId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteSupportIntegration(client, supportId) {
    const url = `/api/v1/integration/support/${supportId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
