// @chatbotkit/sdk/dist/cjs/bot/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BotClient = void 0;
const client_js_1 = require("../client.cjs");
const index_js_1 = require("./session/index.cjs");
const v1_js_1 = require("./v1.cjs");
class BotClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
        this.session = new index_js_1.BotSessionClient(options);
    }
    list(request) {
        return (0, v1_js_1.listBots)(this, request);
    }
    fetch(botId) {
        return (0, v1_js_1.fetchBot)(this, botId);
    }
    create(request) {
        return (0, v1_js_1.createBot)(this, request);
    }
    update(botId, request) {
        return (0, v1_js_1.updateBot)(this, botId, request);
    }
    delete(botId) {
        return (0, v1_js_1.deleteBot)(this, botId);
    }
    upvote(botId, request) {
        return (0, v1_js_1.upvoteBot)(this, botId, request);
    }
    downvote(botId, request) {
        return (0, v1_js_1.downvoteBot)(this, botId, request);
    }
}
exports.BotClient = BotClient;
exports.default = BotClient;
