// @chatbotkit/sdk/dist/cjs/bot/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.downvoteBot = exports.upvoteBot = exports.deleteBot = exports.updateBot = exports.createBot = exports.fetchBot = exports.listBots = void 0;
const v1_js_1 = require("../model/v1.cjs");
function listBots(client, request) {
    let url = `/api/v1/bot/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listBots = listBots;
async function fetchBot(client, botId) {
    const url = `/api/v1/bot/${botId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchBot = fetchBot;
async function createBot(client, request) {
    const url = `/api/v1/bot/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.createBot = createBot;
async function updateBot(client, botId, request) {
    const url = `/api/v1/bot/${botId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.updateBot = updateBot;
async function deleteBot(client, botId) {
    const url = `/api/v1/bot/${botId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteBot = deleteBot;
async function upvoteBot(client, botId, request) {
    const url = `/api/v1/bot/${botId}/upvote`;
    const response = await client.clientFetch(url, {
        record: {
            value: 100,
            ...request,
        },
    });
    return response;
}
exports.upvoteBot = upvoteBot;
async function downvoteBot(client, botId, request) {
    const url = `/api/v1/bot/${botId}/downvote`;
    const response = await client.clientFetch(url, {
        record: {
            value: -100,
            ...request,
        },
    });
    return response;
}
exports.downvoteBot = downvoteBot;
