// @chatbotkit/sdk/dist/cjs/integration/messenger/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessengerIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class MessengerIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listMessengerIntegrations)(this, request);
    }
    fetch(messengerId) {
        return (0, v1_js_1.fetchMessengerIntegration)(this, messengerId);
    }
    create(request) {
        return (0, v1_js_1.createMessengerIntegration)(this, request);
    }
    update(messengerId, request) {
        return (0, v1_js_1.updateMessengerIntegration)(this, messengerId, request);
    }
    delete(messengerId) {
        return (0, v1_js_1.deleteMessengerIntegration)(this, messengerId);
    }
    setup(messengerId) {
        return (0, v1_js_1.setupMessengerIntegration)(this, messengerId);
    }
}
exports.MessengerIntegrationClient = MessengerIntegrationClient;
exports.default = MessengerIntegrationClient;
