// @chatbotkit/sdk/dist/esm/dataset/record/v1.js
export function listDatasetRecords(client, datasetId, request) {
    let url = `/api/v1/dataset/${datasetId}/record/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchDatasetRecord(client, datasetId, recordId) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createDatasetRecord(client, datasetId, request) {
    const url = `/api/v1/dataset/${datasetId}/record/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateDatasetRecord(client, datasetId, recordId, request) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteDatasetRecord(client, datasetId, recordId) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
