// @chatbotkit/sdk/dist/esm/integration/whatsapp/v1.js
import { buildModelString } from '../../model/v1.js';
export function listWhatsAppIntegrations(client, request) {
    let url = `/api/v1/integration/whatsapp/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createWhatsAppIntegration(client, request) {
    const url = `/api/v1/integration/whatsapp/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? buildModelString(request.model) : undefined,
        },
    });
    return response;
}
export async function updateWhatsAppIntegration(client, whatsappId, request) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? buildModelString(request.model) : undefined,
        },
    });
    return response;
}
export async function deleteWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
export async function setupWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/setup`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
