// @chatbotkit/sdk/dist/esm/skillset/ability/v1.js
export function listSkillsetAbilities(client, skillsetId, request) {
    let url = `/api/v1/skillset/${skillsetId}/ability/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchSkillsetAbility(client, skillsetId, abilityId) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createSkillsetAbility(client, skillsetId, request) {
    const url = `/api/v1/skillset/${skillsetId}/ability/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateSkillsetAbility(client, skillsetId, abilityId, request) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteSkillsetAbility(client, skillsetId, abilityId) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
