// @chatbotkit/sdk/dist/esm/secret/v1.js
export function listSecrets(client, request) {
    let url = `/api/v1/secret/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchSecret(client, secretId) {
    const url = `/api/v1/secret/${secretId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createSecret(client, request) {
    const url = `/api/v1/secret/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateSecret(client, secretId, request) {
    const url = `/api/v1/secret/${secretId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteSecret(client, secretId) {
    const url = `/api/v1/secret/${secretId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
