// @chatbotkit/sdk/dist/esm/integration/sitemap/v1.js
export function listSitemapIntegrations(client, request) {
    let url = `/api/v1/integration/sitemap/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createSitemapIntegration(client, request) {
    const url = `/api/v1/integration/sitemap/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateSitemapIntegration(client, sitemapId, request) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
export async function syncSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/sync`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
