// @chatbotkit/sdk/dist/cjs/skillset/ability/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkillsetAbilityClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class SkillsetAbilityClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(skillsetId, request) {
        return (0, v1_js_1.listSkillsetAbilities)(this, skillsetId, request);
    }
    fetch(skillsetId, abilityId) {
        return (0, v1_js_1.fetchSkillsetAbility)(this, skillsetId, abilityId);
    }
    create(skillsetId, request) {
        return (0, v1_js_1.createSkillsetAbility)(this, skillsetId, request);
    }
    update(skillsetId, abilityId, request) {
        return (0, v1_js_1.updateSkillsetAbility)(this, skillsetId, abilityId, request);
    }
    delete(skillsetId, abilityId) {
        return (0, v1_js_1.deleteSkillsetAbility)(this, skillsetId, abilityId);
    }
}
exports.SkillsetAbilityClient = SkillsetAbilityClient;
exports.default = SkillsetAbilityClient;
