// @chatbotkit/sdk/dist/cjs/skillset/ability/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteSkillsetAbility = exports.updateSkillsetAbility = exports.createSkillsetAbility = exports.fetchSkillsetAbility = exports.listSkillsetAbilities = void 0;
function listSkillsetAbilities(client, skillsetId, request) {
    let url = `/api/v1/skillset/${skillsetId}/ability/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listSkillsetAbilities = listSkillsetAbilities;
async function fetchSkillsetAbility(client, skillsetId, abilityId) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchSkillsetAbility = fetchSkillsetAbility;
async function createSkillsetAbility(client, skillsetId, request) {
    const url = `/api/v1/skillset/${skillsetId}/ability/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createSkillsetAbility = createSkillsetAbility;
async function updateSkillsetAbility(client, skillsetId, abilityId, request) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateSkillsetAbility = updateSkillsetAbility;
async function deleteSkillsetAbility(client, skillsetId, abilityId) {
    const url = `/api/v1/skillset/${skillsetId}/ability/${abilityId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteSkillsetAbility = deleteSkillsetAbility;
