// @chatbotkit/sdk/dist/cjs/dataset/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetClient = void 0;
const client_js_1 = require("../client.cjs");
const index_js_1 = require("./file/index.cjs");
const index_js_2 = require("./record/index.cjs");
const v1_js_1 = require("./v1.cjs");
class DatasetClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
        this.file = new index_js_1.DatasetFileClient(options);
        this.record = new index_js_2.DatasetRecordClient(options);
    }
    list(request) {
        return (0, v1_js_1.listDatasets)(this, request);
    }
    fetch(datasetId) {
        return (0, v1_js_1.fetchDataset)(this, datasetId);
    }
    create(request) {
        return (0, v1_js_1.createDataset)(this, request);
    }
    update(datasetId, request) {
        return (0, v1_js_1.updateDataset)(this, datasetId, request);
    }
    delete(datasetId) {
        return (0, v1_js_1.deleteDataset)(this, datasetId);
    }
    search(datasetId, search) {
        return (0, v1_js_1.searchDataset)(this, datasetId, search);
    }
}
exports.DatasetClient = DatasetClient;
exports.default = DatasetClient;
