// @chatbotkit/sdk/dist/cjs/dataset/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.searchDataset = exports.deleteDataset = exports.updateDataset = exports.createDataset = exports.fetchDataset = exports.listDatasets = void 0;
function listDatasets(client, request) {
    let url = `/api/v1/dataset/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listDatasets = listDatasets;
async function fetchDataset(client, datasetId) {
    const url = `/api/v1/dataset/${datasetId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchDataset = fetchDataset;
async function createDataset(client, request) {
    const url = `/api/v1/dataset/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createDataset = createDataset;
async function updateDataset(client, datasetId, request) {
    const url = `/api/v1/dataset/${datasetId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateDataset = updateDataset;
async function deleteDataset(client, datasetId) {
    const url = `/api/v1/dataset/${datasetId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteDataset = deleteDataset;
async function searchDataset(client, datasetId, search) {
    const url = `/api/v1/dataset/${datasetId}/search`;
    const response = await client.clientFetch(url, {
        record: {
            search,
        },
    });
    return response;
}
exports.searchDataset = searchDataset;
