// @chatbotkit/sdk/dist/esm/partner/user/v1.js
export function listPartnerUsers(client, request) {
    let url = `/api/v1/partner/user/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchPartnerUser(client, userId) {
    const url = `/api/v1/partner/user/${userId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createPartnerUser(client, request) {
    const url = `/api/v1/partner/user/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updatePartnerUser(client, userId, request) {
    const url = `/api/v1/partner/user/${userId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deletePartnerUser(client, userId) {
    const url = `/api/v1/partner/user/${userId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
