// @chatbotkit/sdk/dist/esm/integration/widget/v1.js
export function listWidgetIntegrations(client, request) {
    let url = `/api/v1/integration/widget/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchWidgetIntegration(client, widgetId) {
    const url = `/api/v1/integration/widget/${widgetId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createWidgetIntegration(client, request) {
    const url = `/api/v1/integration/widget/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateWidgetIntegration(client, widgetId, request) {
    const url = `/api/v1/integration/widget/${widgetId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteWidgetIntegration(client, widgetId) {
    const url = `/api/v1/integration/widget/${widgetId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
