// @chatbotkit/sdk/dist/cjs/integration/slack/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SlackIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class SlackIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listSlackIntegrations)(this, request);
    }
    fetch(slackId) {
        return (0, v1_js_1.fetchSlackIntegration)(this, slackId);
    }
    create(request) {
        return (0, v1_js_1.createSlackIntegration)(this, request);
    }
    update(slackId, request) {
        return (0, v1_js_1.updateSlackIntegration)(this, slackId, request);
    }
    delete(slackId) {
        return (0, v1_js_1.deleteSlackIntegration)(this, slackId);
    }
    setup(slackId) {
        return (0, v1_js_1.setupSlackIntegration)(this, slackId);
    }
}
exports.SlackIntegrationClient = SlackIntegrationClient;
exports.default = SlackIntegrationClient;
