// @chatbotkit/sdk/dist/cjs/integration/slack/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupSlackIntegration = exports.deleteSlackIntegration = exports.updateSlackIntegration = exports.createSlackIntegration = exports.fetchSlackIntegration = exports.listSlackIntegrations = void 0;
const v1_js_1 = require("../../model/v1.cjs");
function listSlackIntegrations(client, request) {
    let url = `/api/v1/integration/slack/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listSlackIntegrations = listSlackIntegrations;
async function fetchSlackIntegration(client, slackId) {
    const url = `/api/v1/integration/slack/${slackId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchSlackIntegration = fetchSlackIntegration;
async function createSlackIntegration(client, request) {
    const url = `/api/v1/integration/slack/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.createSlackIntegration = createSlackIntegration;
async function updateSlackIntegration(client, slackId, request) {
    const url = `/api/v1/integration/slack/${slackId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.updateSlackIntegration = updateSlackIntegration;
async function deleteSlackIntegration(client, slackId) {
    const url = `/api/v1/integration/slack/${slackId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteSlackIntegration = deleteSlackIntegration;
async function setupSlackIntegration(client, slackId) {
    const url = `/api/v1/integration/slack/${slackId}/setup`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.setupSlackIntegration = setupSlackIntegration;
