// @chatbotkit/sdk/dist/esm/dataset/index.js
import { ChatBotKitClient } from '../client.js';
import { DatasetFileClient } from './file/index.js';
import { DatasetRecordClient } from './record/index.js';
import { createDataset, deleteDataset, fetchDataset, listDatasets, searchDataset, updateDataset, } from './v1.js';
export class DatasetClient extends ChatBotKitClient {
    constructor(options) {
        super(options);
        this.file = new DatasetFileClient(options);
        this.record = new DatasetRecordClient(options);
    }
    list(request) {
        return listDatasets(this, request);
    }
    fetch(datasetId) {
        return fetchDataset(this, datasetId);
    }
    create(request) {
        return createDataset(this, request);
    }
    update(datasetId, request) {
        return updateDataset(this, datasetId, request);
    }
    delete(datasetId) {
        return deleteDataset(this, datasetId);
    }
    search(datasetId, search) {
        return searchDataset(this, datasetId, search);
    }
}
export default DatasetClient;
