// @chatbotkit/sdk/dist/esm/dataset/v1.js
export function listDatasets(client, request) {
    let url = `/api/v1/dataset/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchDataset(client, datasetId) {
    const url = `/api/v1/dataset/${datasetId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createDataset(client, request) {
    const url = `/api/v1/dataset/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateDataset(client, datasetId, request) {
    const url = `/api/v1/dataset/${datasetId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteDataset(client, datasetId) {
    const url = `/api/v1/dataset/${datasetId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
export async function searchDataset(client, datasetId, search) {
    const url = `/api/v1/dataset/${datasetId}/search`;
    const response = await client.clientFetch(url, {
        record: {
            search,
        },
    });
    return response;
}
