// @chatbotkit/sdk/dist/esm/integration/notion/v1.js
export function listNotionIntegrations(client, request) {
    let url = `/api/v1/integration/notion/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchNotionIntegration(client, notionId) {
    const url = `/api/v1/integration/notion/${notionId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createNotionIntegration(client, request) {
    const url = `/api/v1/integration/notion/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateNotionIntegration(client, notionId, request) {
    const url = `/api/v1/integration/notion/${notionId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteNotionIntegration(client, notionId) {
    const url = `/api/v1/integration/notion/${notionId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
export async function syncNotionIntegration(client, notionId) {
    const url = `/api/v1/integration/notion/${notionId}/sync`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
