// @chatbotkit/sdk/dist/esm/integration/extract/v1.js
export function listExtractIntegrations(client, request) {
    let url = `/api/v1/integration/extract/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchExtractIntegration(client, extractId) {
    const url = `/api/v1/integration/extract/${extractId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createExtractIntegration(client, request) {
    const url = `/api/v1/integration/extract/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateExtractIntegration(client, extractId, request) {
    const url = `/api/v1/integration/extract/${extractId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteExtractIntegration(client, extractId) {
    const url = `/api/v1/integration/extract/${extractId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
