// @chatbotkit/sdk/dist/cjs/skillset/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteSkillset = exports.updateSkillset = exports.createSkillset = exports.fetchSkillset = exports.listSkillsets = void 0;
function listSkillsets(client, request) {
    let url = `/api/v1/skillset/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listSkillsets = listSkillsets;
async function fetchSkillset(client, skillsetId) {
    const url = `/api/v1/skillset/${skillsetId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchSkillset = fetchSkillset;
async function createSkillset(client, request) {
    const url = `/api/v1/skillset/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createSkillset = createSkillset;
async function updateSkillset(client, skillsetId, request) {
    const url = `/api/v1/skillset/${skillsetId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateSkillset = updateSkillset;
async function deleteSkillset(client, skillsetId) {
    const url = `/api/v1/skillset/${skillsetId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteSkillset = deleteSkillset;
