// @chatbotkit/sdk/dist/cjs/contact/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteContact = exports.updateContact = exports.createContact = exports.fetchContact = exports.listContacts = void 0;
function listContacts(client, request) {
    let url = `/api/v1/contact/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listContacts = listContacts;
async function fetchContact(client, contactId) {
    const url = `/api/v1/contact/${contactId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchContact = fetchContact;
async function createContact(client, request) {
    const url = `/api/v1/contact/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createContact = createContact;
async function updateContact(client, contactId, request) {
    const url = `/api/v1/contact/${contactId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateContact = updateContact;
async function deleteContact(client, contactId) {
    const url = `/api/v1/contact/${contactId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteContact = deleteContact;
