// @chatbotkit/sdk/dist/cjs/integration/extract/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteExtractIntegration = exports.updateExtractIntegration = exports.createExtractIntegration = exports.fetchExtractIntegration = exports.listExtractIntegrations = void 0;
function listExtractIntegrations(client, request) {
    let url = `/api/v1/integration/extract/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listExtractIntegrations = listExtractIntegrations;
async function fetchExtractIntegration(client, extractId) {
    const url = `/api/v1/integration/extract/${extractId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchExtractIntegration = fetchExtractIntegration;
async function createExtractIntegration(client, request) {
    const url = `/api/v1/integration/extract/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createExtractIntegration = createExtractIntegration;
async function updateExtractIntegration(client, extractId, request) {
    const url = `/api/v1/integration/extract/${extractId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateExtractIntegration = updateExtractIntegration;
async function deleteExtractIntegration(client, extractId) {
    const url = `/api/v1/integration/extract/${extractId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteExtractIntegration = deleteExtractIntegration;
