// @chatbotkit/sdk/dist/esm/integration/index.js
import { ChatBotKitClient } from '../client.js';
import { DiscordIntegrationClient } from './discord/index.js';
import { ExtractIntegrationClient } from './extract/index.js';
import { MessengerIntegrationClient } from './messenger/index.js';
import { NotionIntegrationClient } from './notion/index.js';
import { SitemapIntegrationClient } from './sitemap/index.js';
import { SlackIntegrationClient } from './slack/index.js';
import { SupportIntegrationClient } from './support/index.js';
import { TelegramIntegrationClient } from './telegram/index.js';
import { WhatsAppIntegrationClient } from './whatsapp/index.js';
import { WidgetIntegrationClient } from './widget/index.js';
export class IntegrationClient extends ChatBotKitClient {
    constructor(options) {
        super(options);
        this.widget = new WidgetIntegrationClient(options);
        this.slack = new SlackIntegrationClient(options);
        this.discord = new DiscordIntegrationClient(options);
        this.whatsapp = new WhatsAppIntegrationClient(options);
        this.messenger = new MessengerIntegrationClient(options);
        this.telegram = new TelegramIntegrationClient(options);
        this.sitemap = new SitemapIntegrationClient(options);
        this.notion = new NotionIntegrationClient(options);
        this.support = new SupportIntegrationClient(options);
        this.extract = new ExtractIntegrationClient(options);
    }
}
export default IntegrationClient;
