// @chatbotkit/sdk/dist/cjs/secret/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteSecret = exports.updateSecret = exports.createSecret = exports.fetchSecret = exports.listSecrets = void 0;
function listSecrets(client, request) {
    let url = `/api/v1/secret/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listSecrets = listSecrets;
async function fetchSecret(client, secretId) {
    const url = `/api/v1/secret/${secretId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchSecret = fetchSecret;
async function createSecret(client, request) {
    const url = `/api/v1/secret/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createSecret = createSecret;
async function updateSecret(client, secretId, request) {
    const url = `/api/v1/secret/${secretId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateSecret = updateSecret;
async function deleteSecret(client, secretId) {
    const url = `/api/v1/secret/${secretId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteSecret = deleteSecret;
