// @chatbotkit/sdk/dist/cjs/integration/sitemap/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SitemapIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class SitemapIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listSitemapIntegrations)(this, request);
    }
    fetch(sitemapId) {
        return (0, v1_js_1.fetchSitemapIntegration)(this, sitemapId);
    }
    create(request) {
        return (0, v1_js_1.createSitemapIntegration)(this, request);
    }
    update(sitemapId, request) {
        return (0, v1_js_1.updateSitemapIntegration)(this, sitemapId, request);
    }
    delete(sitemapId) {
        return (0, v1_js_1.deleteSitemapIntegration)(this, sitemapId);
    }
    sync(sitemapId) {
        return (0, v1_js_1.syncSitemapIntegration)(this, sitemapId);
    }
}
exports.SitemapIntegrationClient = SitemapIntegrationClient;
exports.default = SitemapIntegrationClient;
