// @chatbotkit/sdk/dist/cjs/integration/sitemap/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.syncSitemapIntegration = exports.deleteSitemapIntegration = exports.updateSitemapIntegration = exports.createSitemapIntegration = exports.fetchSitemapIntegration = exports.listSitemapIntegrations = void 0;
function listSitemapIntegrations(client, request) {
    let url = `/api/v1/integration/sitemap/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listSitemapIntegrations = listSitemapIntegrations;
async function fetchSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchSitemapIntegration = fetchSitemapIntegration;
async function createSitemapIntegration(client, request) {
    const url = `/api/v1/integration/sitemap/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createSitemapIntegration = createSitemapIntegration;
async function updateSitemapIntegration(client, sitemapId, request) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateSitemapIntegration = updateSitemapIntegration;
async function deleteSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteSitemapIntegration = deleteSitemapIntegration;
async function syncSitemapIntegration(client, sitemapId) {
    const url = `/api/v1/integration/sitemap/${sitemapId}/sync`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.syncSitemapIntegration = syncSitemapIntegration;
