// @chatbotkit/sdk/dist/cjs/partner/user/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deletePartnerUser = exports.updatePartnerUser = exports.createPartnerUser = exports.fetchPartnerUser = exports.listPartnerUsers = void 0;
function listPartnerUsers(client, request) {
    let url = `/api/v1/partner/user/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listPartnerUsers = listPartnerUsers;
async function fetchPartnerUser(client, userId) {
    const url = `/api/v1/partner/user/${userId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchPartnerUser = fetchPartnerUser;
async function createPartnerUser(client, request) {
    const url = `/api/v1/partner/user/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createPartnerUser = createPartnerUser;
async function updatePartnerUser(client, userId, request) {
    const url = `/api/v1/partner/user/${userId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updatePartnerUser = updatePartnerUser;
async function deletePartnerUser(client, userId) {
    const url = `/api/v1/partner/user/${userId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deletePartnerUser = deletePartnerUser;
