// @chatbotkit/sdk/dist/cjs/integration/whatsapp/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WhatsAppIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class WhatsAppIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listWhatsAppIntegrations)(this, request);
    }
    fetch(whatsappId) {
        return (0, v1_js_1.fetchWhatsAppIntegration)(this, whatsappId);
    }
    create(request) {
        return (0, v1_js_1.createWhatsAppIntegration)(this, request);
    }
    update(whatsappId, request) {
        return (0, v1_js_1.updateWhatsAppIntegration)(this, whatsappId, request);
    }
    delete(whatsappId) {
        return (0, v1_js_1.deleteWhatsAppIntegration)(this, whatsappId);
    }
    setup(whatsappId) {
        return (0, v1_js_1.setupWhatsAppIntegration)(this, whatsappId);
    }
}
exports.WhatsAppIntegrationClient = WhatsAppIntegrationClient;
exports.default = WhatsAppIntegrationClient;
