// @chatbotkit/sdk/dist/cjs/integration/whatsapp/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupWhatsAppIntegration = exports.deleteWhatsAppIntegration = exports.updateWhatsAppIntegration = exports.createWhatsAppIntegration = exports.fetchWhatsAppIntegration = exports.listWhatsAppIntegrations = void 0;
const v1_js_1 = require("../../model/v1.cjs");
function listWhatsAppIntegrations(client, request) {
    let url = `/api/v1/integration/whatsapp/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listWhatsAppIntegrations = listWhatsAppIntegrations;
async function fetchWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchWhatsAppIntegration = fetchWhatsAppIntegration;
async function createWhatsAppIntegration(client, request) {
    const url = `/api/v1/integration/whatsapp/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.createWhatsAppIntegration = createWhatsAppIntegration;
async function updateWhatsAppIntegration(client, whatsappId, request) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
            model: request.model ? (0, v1_js_1.buildModelString)(request.model) : undefined,
        },
    });
    return response;
}
exports.updateWhatsAppIntegration = updateWhatsAppIntegration;
async function deleteWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteWhatsAppIntegration = deleteWhatsAppIntegration;
async function setupWhatsAppIntegration(client, whatsappId) {
    const url = `/api/v1/integration/whatsapp/${whatsappId}/setup`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.setupWhatsAppIntegration = setupWhatsAppIntegration;
