// @chatbotkit/sdk/dist/cjs/dataset/record/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetRecordClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class DatasetRecordClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(datasetId, request) {
        return (0, v1_js_1.listDatasetRecords)(this, datasetId, request);
    }
    fetch(datasetId, recordId) {
        return (0, v1_js_1.fetchDatasetRecord)(this, datasetId, recordId);
    }
    create(datasetId, request) {
        return (0, v1_js_1.createDatasetRecord)(this, datasetId, request);
    }
    update(datasetId, recordId, request) {
        return (0, v1_js_1.updateDatasetRecord)(this, datasetId, recordId, request);
    }
    delete(datasetId, recordId) {
        return (0, v1_js_1.deleteDatasetRecord)(this, datasetId, recordId);
    }
}
exports.DatasetRecordClient = DatasetRecordClient;
exports.default = DatasetRecordClient;
