// @chatbotkit/sdk/dist/cjs/dataset/record/v1.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteDatasetRecord = exports.updateDatasetRecord = exports.createDatasetRecord = exports.fetchDatasetRecord = exports.listDatasetRecords = void 0;
function listDatasetRecords(client, datasetId, request) {
    let url = `/api/v1/dataset/${datasetId}/record/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
exports.listDatasetRecords = listDatasetRecords;
async function fetchDatasetRecord(client, datasetId, recordId) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
exports.fetchDatasetRecord = fetchDatasetRecord;
async function createDatasetRecord(client, datasetId, request) {
    const url = `/api/v1/dataset/${datasetId}/record/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.createDatasetRecord = createDatasetRecord;
async function updateDatasetRecord(client, datasetId, recordId, request) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
exports.updateDatasetRecord = updateDatasetRecord;
async function deleteDatasetRecord(client, datasetId, recordId) {
    const url = `/api/v1/dataset/${datasetId}/record/${recordId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
exports.deleteDatasetRecord = deleteDatasetRecord;
