// @chatbotkit/sdk/dist/esm/skillset/v1.js
export function listSkillsets(client, request) {
    let url = `/api/v1/skillset/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchSkillset(client, skillsetId) {
    const url = `/api/v1/skillset/${skillsetId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createSkillset(client, request) {
    const url = `/api/v1/skillset/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateSkillset(client, skillsetId, request) {
    const url = `/api/v1/skillset/${skillsetId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteSkillset(client, skillsetId) {
    const url = `/api/v1/skillset/${skillsetId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
