// @chatbotkit/sdk/dist/cjs/integration/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationClient = void 0;
const client_js_1 = require("../client.cjs");
const index_js_1 = require("./discord/index.cjs");
const index_js_2 = require("./extract/index.cjs");
const index_js_3 = require("./messenger/index.cjs");
const index_js_4 = require("./notion/index.cjs");
const index_js_5 = require("./sitemap/index.cjs");
const index_js_6 = require("./slack/index.cjs");
const index_js_7 = require("./support/index.cjs");
const index_js_8 = require("./telegram/index.cjs");
const index_js_9 = require("./whatsapp/index.cjs");
const index_js_10 = require("./widget/index.cjs");
class IntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
        this.widget = new index_js_10.WidgetIntegrationClient(options);
        this.slack = new index_js_6.SlackIntegrationClient(options);
        this.discord = new index_js_1.DiscordIntegrationClient(options);
        this.whatsapp = new index_js_9.WhatsAppIntegrationClient(options);
        this.messenger = new index_js_3.MessengerIntegrationClient(options);
        this.telegram = new index_js_8.TelegramIntegrationClient(options);
        this.sitemap = new index_js_5.SitemapIntegrationClient(options);
        this.notion = new index_js_4.NotionIntegrationClient(options);
        this.support = new index_js_7.SupportIntegrationClient(options);
        this.extract = new index_js_2.ExtractIntegrationClient(options);
    }
}
exports.IntegrationClient = IntegrationClient;
exports.default = IntegrationClient;
