// @chatbotkit/sdk/dist/esm/contact/v1.js
export function listContacts(client, request) {
    let url = `/api/v1/contact/list`;
    const response = client.clientFetch(url, { query: request });
    return response;
}
export async function fetchContact(client, contactId) {
    const url = `/api/v1/contact/${contactId}/fetch`;
    const response = await client.clientFetch(url);
    return response;
}
export async function createContact(client, request) {
    const url = `/api/v1/contact/create`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function updateContact(client, contactId, request) {
    const url = `/api/v1/contact/${contactId}/update`;
    const response = await client.clientFetch(url, {
        record: {
            ...request,
        },
    });
    return response;
}
export async function deleteContact(client, contactId) {
    const url = `/api/v1/contact/${contactId}/delete`;
    const response = await client.clientFetch(url, {
        record: {},
    });
    return response;
}
