// @chatbotkit/fetch/dist/esm/index.js
import { Blob, FormData, fetch } from 'node-fetch-native';
export { fetch, Blob, FormData };
if (typeof globalThis.ReadableStream === 'function' &&
    typeof globalThis.ReadableStream.prototype[Symbol.asyncIterator] !==
        'function') {
    globalThis.ReadableStream.prototype[Symbol.asyncIterator] = function () {
        const reader = this.getReader();
        return {
            next: () => reader.read(),
            return: () => {
                reader.releaseLock();
            },
        };
    };
}
export const ABORT_ERROR_NAME = 'AbortError';
export const TIMEOUT_ERROR_NAME = 'TimeoutError';
function anySignal(signals) {
    const controller = new AbortController();
    for (const signal of signals) {
        if (signal.aborted) {
            controller.abort(signal.reason);
            return signal;
        }
        signal.addEventListener('abort', () => controller.abort(signal.reason), {
            signal: controller.signal,
        });
    }
    return controller.signal;
}
export class FetchError extends Error {
    constructor(message, code, url, request, response) {
        super(message);
        this.code = code || response.status;
        this.url = url;
        this.request = request;
        this.response = response;
    }
}
export function withTimeout(fetch, defaultOptions) {
    return async function fetchWithTimeout(url, options) {
        const timeout = options?.timeout ?? defaultOptions?.timeout ?? 30000;
        let signal;
        let handler;
        let isTimeOutAbort = false;
        if (timeout > 0 && timeout !== Infinity) {
            const abortController = new AbortController();
            handler = setTimeout(() => {
                isTimeOutAbort = true;
                abortController.abort(TIMEOUT_ERROR_NAME);
            }, timeout);
            signal = options?.signal
                ? anySignal([abortController.signal, options.signal])
                : abortController.signal;
        }
        else {
            signal = options?.signal;
        }
        let response;
        try {
            response = await fetch(url, {
                ...options,
                signal,
            });
        }
        catch (error) {
            if ([error.name, error.message].includes(ABORT_ERROR_NAME)) {
                if (isTimeOutAbort) {
                    throw new Error(TIMEOUT_ERROR_NAME);
                }
            }
            throw error;
        }
        finally {
            clearTimeout(handler);
        }
        return response;
    };
}
export function withRetry(fetch, defaultOptions) {
    return async function fetchWithRetry(url, options) {
        const retries = options?.retries ?? defaultOptions?.retries ?? 5;
        const retryDelay = options?.retryDelay ?? defaultOptions?.retryDelay ?? 250;
        const retryTimeout = options?.retryTimeout ?? defaultOptions?.retryTimeout ?? false;
        const retryStatuses = options?.retryStatuses ??
            defaultOptions?.retryStatuses ?? [429, 500, 502, 503, 504];
        let response;
        try {
            response = await fetch(url, { ...options });
            if (!response.ok) {
                if (retryStatuses.includes(response.status)) {
                    throw new FetchError(`Request failed`, response.status, url, options || {}, response);
                }
                else {
                    return response;
                }
            }
            return response;
        }
        catch (error) {
            switch (true) {
                case [error.name, error.message].includes(TIMEOUT_ERROR_NAME) &&
                    !retryTimeout:
                    {
                        throw error;
                    }
                case retries === 0: {
                    if (response) {
                        return response;
                    }
                    else {
                        throw error;
                    }
                }
            }
            await new Promise((resolve) => setTimeout(resolve, retryDelay));
            return await fetchWithRetry(url, {
                ...options,
                retries: retries - 1,
                retryDelay: retryDelay * 2,
            });
        }
    };
}
export async function* jsonl(body) {
    const decoder = new TextDecoder();
    let previous = '';
    for await (const chunk of body) {
        previous += decoder.decode(chunk);
        let eolIndex;
        while ((eolIndex = previous.indexOf('\n')) >= 0) {
            const line = previous.slice(0, eolIndex + 1);
            if (line) {
                yield JSON.parse(line);
            }
            previous = previous.slice(eolIndex + 1);
        }
    }
    if (previous.trim().length > 0) {
        yield JSON.parse(previous);
    }
}
