// @chatbotkit/sdk/dist/cjs/integration/notion/index.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotionIntegrationClient = void 0;
const client_js_1 = require("../../client.cjs");
const v1_js_1 = require("./v1.cjs");
class NotionIntegrationClient extends client_js_1.ChatBotKitClient {
    constructor(options) {
        super(options);
    }
    list(request) {
        return (0, v1_js_1.listNotionIntegrations)(this, request);
    }
    fetch(notionId) {
        return (0, v1_js_1.fetchNotionIntegration)(this, notionId);
    }
    create(request) {
        return (0, v1_js_1.createNotionIntegration)(this, request);
    }
    update(notionId, request) {
        return (0, v1_js_1.updateNotionIntegration)(this, notionId, request);
    }
    delete(notionId) {
        return (0, v1_js_1.deleteNotionIntegration)(this, notionId);
    }
    sync(notionId) {
        return (0, v1_js_1.syncNotionIntegration)(this, notionId);
    }
}
exports.NotionIntegrationClient = NotionIntegrationClient;
exports.default = NotionIntegrationClient;
